src/vivarium_gates_nutrition_optimization/components/pregnancy.py

```diff
@@ -38,36 +40,82 @@ def columns_created(self):
             "pregnancy_duration",
         ] + self.new_children.columns_created
 
-    def setup(self, builder: Builder):
-        super(BaseDiseaseState, self).setup(builder)
+    @property
+    def sub_components(self):
+        return super().sub_components + [self.new_children]
 
+    @property
+    def initialization_requirements(self) -> Dict[str, List[str]]:
+        return {
+            "requires_columns": [self.model],
+            "requires_values": ["birth_outcome_probabilities"],
+            "requires_streams": [],
+        }
+
+    def setup(self, builder: Builder):
+        """Performs this component's simulation setup.
+
+        Parameters
+        ----------
+        builder : `engine.Builder`
+            Interface to several simulation tools.
+        """
+        super(DiseaseState, self).setup(builder)
```

Comment:
Why did this change from `super(BaseDiseaseState, self).setup(builder)`?

Reply:
BaseDiseaseState actually doesn't have a distinct setup method, so in this case super(DiseaseState, self) == super() and super(BaseDiseaseState, self) are all equivalent. That said, I shoud just change it to super()